### PR TITLE
Use uber-go/zap as our new logger

### PR DESCRIPTION
**Pull request by MDr164**
This logger is not only very fast but also allows combined logging to the console and to a file while each logging backend can be configured individually. The syntax and usage is also a drop-in replacement for the standard library log package.

### CODE REVIEW

File: pkg/bmc/cert/load.go [hunk 1]
```diff
@@ -227,7 +227,7 @@ func (m *Manager) renew() (*tls.Certificate, error) {
 		}
 	}
 	if !handled {
-		return nil, fmt.Errorf("No DNS01 ACME handler could handle challenge")
+		return nil, fmt.Errorf("fo DNS01 ACME handler could handle challenge")
```
Comment (walterchris) [inline pkg/bmc/cert/load.go:230]: fo?

File: pkg/bmc/interface.go [hunk 2]
```diff
@@ -125,9 +124,15 @@ func ipv6LinkFixer(iface string) {
 			}
 		}
 		if !found {
-			log.Printf("No link-local IPv6 address found for %s, resetting interface", iface)
-			setLinkDown(iface)
-			setLinkUp(iface)
+			log.Errorf("No link-local IPv6 address found for %s, resetting interface", iface)
+			err = setLinkDown(iface)
```
Comment (walterchris) [inline pkg/bmc/interface.go:128]: errors don't matter?

**Review by walterchris — CHANGES_REQUESTED** [review]

File: pkg/bmc/cert/load.go (hunk 1 above)
Comment (MDr164) [reply pkg/bmc/cert/load.go]: Fixed

File: pkg/bmc/interface.go (hunk 2 above)
Comment (MDr164) [reply pkg/bmc/interface.go:128]: Added "continue" to reload the loop

**Review by walterchris — APPROVED** [review]

**Pull request merged**